### PR TITLE
Use base.url for '/' root relative links

### PR DESCRIPTION
**Pull request by ashawley**
As mentioned in #854, html-proofer is indeed handy for finding bad links. It found a number of them, including page anchors.  That's one thing wget can't do.  I didn't fix all the bad links, but did a number of them.  Though, the links on language pages other than english are in a bit of a disorder for the moment.

I was  trying to sprinkle `site.baseurl` around per discussion in #849.  I did use wget to verify hosting the site on a web server with a non-empty baseurl works better.  Should help avoid needing do any JS link hackery.

### CODE REVIEW

File: _overviews/collections/trait-traversable.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ The `foreach` method is meant to traverse all elements of the collection, and ap
 * **Folds** `foldLeft`, `foldRight`, `/:`, `:\`, `reduceLeft`, `reduceRight` which apply a binary operation to successive elements.
 * **Specific folds** `sum`, `product`, `min`, `max`, which work on collections of specific types (numeric or comparable).
 * **String** operations `mkString`, `addString`, `stringPrefix`, which give alternative ways of converting a collection to a string.
-* **View** operations, consisting of two overloaded variants of the `view` method. A view is a collection that's evaluated lazily. You'll learn more about views in [later](#Views).
+* **View** operations, consisting of two overloaded variants of the `view` method. A view is a collection that's evaluated lazily. You'll learn more about views in [later](views.html).
```
Comment (ashawley) [inline _overviews/collections/trait-traversable.md]: Page was separated at some point?

File: _overviews/macros/annotations.md [hunk 2]
```diff
@@ -18,7 +18,7 @@ permalink: /overviews/macros/:title.html
 
 Macro annotations are only available with the macro paradise plugin (in Scala 2.10.x, 2.11.x and 2.12.x alike).
 Their inclusion in official Scala might happen in Scala 2.13, but there is no certainty about it yet.
-Follow the instructions at the ["Macro Paradise"](/overviews/macros/paradise.html) page to download and use our compiler plugin.
+Follow the instructions at the ["Macro Paradise"](paradise.html) page to download and use our compiler plugin.
```
Comment (ashawley) [inline _overviews/macros/annotations.md]: This is a link to a file in the same directory

File: _overviews/macros/blackbox-whitebox.md [hunk 3]
```diff
@@ -35,7 +35,7 @@ compehensibility.
...
 
-This curious feature provides additional flexibility, enabling [fake type providers](http://meta.plasm.us/posts/2013/07/11/fake-type-providers-part-2/), [extended vanilla materialization](/sips/pending/source-locations.html), [fundep materialization](/overviews/macros/implicits.html#fundep-materialization) and [extractor macros](https://github.com/scala/scala/commit/84a335916556cb0fe939d1c51f27d80d9cf980dc), but it also sacrifices clarity - both for humans and for machines.
+This curious feature provides additional flexibility, enabling [fake type providers](http://meta.plasm.us/posts/2013/07/11/fake-type-providers-part-2/), [extended vanilla materialization](/sips/pending/source-locations.html), [fundep materialization]({{ site.baseurl }}/overviews/macros/implicits.html#fundep-materialization) and [extractor macros](https://github.com/scala/scala/commit/84a335916556cb0fe939d1c51f27d80d9cf980dc), but it also sacrifices clarity - both for humans and for machines.
```
Comment (ashawley) [inline _overviews/macros/blackbox-whitebox.md]: I don't fix all SIPS links, but I guess I did here

File: _includes/books.html [hunk 4]
```diff
@@ -5,7 +5,7 @@
       <div class="book-item-header">
         <a href="{{book.link}}">
           <div class="content-img-boook">
-            <img src="{{book.image}}" alt="{{book.title}}">
+            <img src="{{site.baseurl}}{{book.image}}" alt="{{book.title}}">
```
Comment (ashawley) [inline _includes/books.html]: This is the theme of the show

File: _includes/footer.html [hunk 5]
```diff
@@ -5,7 +5,7 @@
           <ul class="{{item.class}}">
             <li><h3>{{item.title}}</h3></li>
             {% for link in item.links %}
-              <li><a href="{{link.url}}">{{link.title}}</a></li>
+              <li><a href="{% if link.url contains '://' %}{{link.url}}{% else %}{{site.baseurl}}{{link.url}}{% endif %}">{{link.title}}</a></li>
```
Comment (ashawley) [inline _includes/footer.html]: I use this absolute link boilerplate a lot

File: _overviews/macros/bundles.md [hunk 6]
```diff
@@ -27,7 +27,7 @@ following reasons:
 1. Being limited to functions makes modularizing complex macros awkward. It's quite typical to see macro logic concentrate in helper
 traits outside macro implementations, turning implementations into trivial wrappers, which just instantiate and call helpers.
 
-2. Moreover, since macro parameters are path-dependent on the macro context, [special incantations](/overviews/macros/overview.html#writing-bigger-macros) are required to wire implementations and helpers together.
+2. Moreover, since macro parameters are path-dependent on the macro context, [special incantations](overview.html#writing-bigger-macros) are required to wire implementations and helpers together.
```
Comment (ashawley) [inline _overviews/macros/bundles.md]: Same working directory again

File: contribute.md [hunk 7]
```diff
@@ -124,7 +124,7 @@ At the moment, a tutorial that can be logically placed on **one** page must be p
     title: YOUR TITLE
     ---
 
-If you have a **multiple-page** tutorial, like in the case of multiple-page overviews, you must both specify an ordering for your document, and a name must be assigned to the collection of tutorial pages. For example, the following header is used for the [Tour of Scala]({{ site.baseurl }}/tutorials) series of tutorial articles:
+If you have a **multiple-page** tutorial, like in the case of multiple-page overviews, you must both specify an ordering for your document, and a name must be assigned to the collection of tutorial pages. For example, the following header is used for the [Tour of Scala]({{ site.baseurl }}/tour/tour-of-scala.html) series of tutorial articles:
```
Comment (ashawley) [inline contribute.md]: The tour moved

File: _tour/tour-of-scala.md [hunk 8]
```diff
@@ -19,7 +19,7 @@ Scala is a modern multi-paradigm programming language designed to express common
...
+Scala is also a functional language in the sense that [every function is a value](unified-types.html). Scala provides a [lightweight syntax](basics.html#functions) for defining anonymous functions, it supports [higher-order functions](higher-order-functions.html), it allows functions to be [nested](nested-functions.html), and supports [currying](currying.html). Scala's [case classes](case-classes.html) and its built-in support for [pattern matching](pattern-matching.html) model algebraic types used in many functional programming languages. [Singleton objects](singleton-objects.html) provide a convenient way to group functions that aren't members of a class.
```
Comment (ashawley) [inline _tour/tour-of-scala.md]: Adding the missing the `s` in `basics.html#functions`

**Comment by heathermiller** [comment]
🎉 🍻 THANK YOU!

**Pull request merged**